File: Projects/Project_Cinema Website/backend/src/main/java/com/cinema_website/backend/service/MovieService.java
package com.cinema_website.backend.service;

import com.cinema_website.backend.dto.MovieDTO;
import com.cinema_website.backend.mapper.MovieMapper;
import com.cinema_website.backend.repository.MovieRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;



@SuppressWarnings("all")
@Service
public class MovieService {

    private final MovieRepository movieRepository;
    private final MovieMapper movieMapper;

    @Autowired
    public MovieService(MovieRepository movieRepository, MovieMapper movieMapper) {
        this.movieRepository = movieRepository;
        this.movieMapper = movieMapper;
    }


    // GET
    public MovieDTO getMovie(int movieId) {
        return movieRepository.getMovieById(movieId);
    }

    public List<MovieDTO> getAllMovies() {
        return movieRepository.getAllMovies();
    }


    // INSERT
    public boolean addMovie(MovieDTO movie) {
        return movieRepository.addMovie(movie);
    }


    // UPDATE
    public boolean updateMovie(int movieId, MovieDTO updatedMovie) {
        return movieRepository.updateMovieById(movieId, updatedMovie);
    }


    // DELETE
    public boolean deleteMovie(int movieId) {
        return movieRepository.deleteMovieById(movieId);
    }


}

File: Projects/Project_Cinema Website/frontend/src/services/MovieService.js
import axios from "axios";

const baseURL = 'http://localhost:8080/api/movies'; // Replace with your actual API endpoint

class MovieService {

    // GET
    static async getMovieById(movieId) {
        return (await axios.get(`${baseURL}/getMovie/${movieId}`)).data;
    }

    static async getAllMovies() {
        return (await axios.get(`${baseURL}/getAllMovies`)).data;
    }

    // POST
    static async addMovie(newMovie) {
        return (await axios.post(`${baseURL}/addMovie`, newMovie)).data;
    }

    // PUT (Update)
    static async updateMovieById(movieId, updatedMovie) {
        return (await axios.put(`${baseURL}/updateMovie/${movieId}`, updatedMovie)).data;
    }

    // DELETE
    static async deleteMovieById(movieId) {
        return (await axios.delete(`${baseURL}/deleteMovie/${movieId}`)).data;
    }
}

export default MovieService;
